vault/external_tests/identity/login_mfa_okta_test.go

```diff
@@ -6,20 +6,56 @@ package identity
 import (
 	"context"
 	"fmt"
+	"os"
 	"reflect"
+	"strings"
 	"testing"
+	"time"
 
 	"github.com/hashicorp/vault/api"
+	"github.com/hashicorp/vault/helper/constants"
+	"github.com/hashicorp/vault/helper/testhelpers"
+	logicaltest "github.com/hashicorp/vault/helper/testhelpers/logical"
 	"github.com/hashicorp/vault/helper/testhelpers/minimal"
 )
 
-var (
-	org_name  = "<okta org name>"
-	api_token = "<okta api token>"
-)
+// To run these tests, set the following env variables:
+// VAULT_ACC=1
+// OKTA_ORG=dev-219337
+// OKTA_API_TOKEN=<generate via web UI, see Confluence for login details>
+// OKTA_USERNAME=test3@example.com
+// OKTA_PASSWORD=<find in 1password>
+//
+// You will need to install the Okta client app on your mobile device and
+// setup MFA in order to use the Okta web UI. This test does not exercise
+// MFA however (which is an enterprise feature), and therefore the test
```

Comment:
This comment was one other thing I ended up unsure about - did you discover for certain that MFA for Okta Auth method is Enterprise? I didn't see that we obviously state that anywhere (although it's confusing with so many different ways to do MFA with Okta).

My rough understanding here is that the Okta auth method will do MFA if your okta org requires it (and it's not bypassed in the auth method config). One complication is that Okta changed their policy engine and it's actually not obvious how to require MFA from API-driven logins anymore since their new policy engine's Authentication Policies seem to be App centric.

I've not fully verified this yet but I now believe [based on this thread ](https://devforum.okta.com/t/cant-get-status-mfa-enroll-when-authenticating-with-api/23506/5)that if your org has a Global Session Policy that requires MFA, and your user has an MFA device enrolled, then this test should trigger MFA in CE and Enterprise.

If you know any of those things to be incorrect @cognifloyd please do let us know! We can verify them when we finish this off but you may already be able to confirm or deny that!